src/Model/RecordLocale.php

```diff
@@ -351,14 +351,20 @@ public function getSourceLocale(): ?Locale
     {
         /** @var DataObject|FluentExtension $record */
         $record = $this->getOriginalRecord();
+        $validationMethod = 'existsInLocale';
 
-        if ($record->existsInLocale($this->getLocale())) {
+        if ($record->hasExtension(FluentVersionedExtension::class)) {
```

Comment:
Maybe `existsInLocale` should check only the CURRENT stage, instead of both stages?

FluentVersionedExtension has this code

```php
    /**
     * Check if this record exists (in either state) in this locale
     *
     * @param string $locale
     * @return bool
     */
    public function existsInLocale($locale = null)
    {
        return $this->isDraftedInLocale($locale) || $this->isPublishedInLocale($locale);
    }
```

maybe it should be this?

```php
    /**
     * Check if this record exists (in either state) in this locale
     *
     * @param string $locale
     * @return bool
     */
    public function existsInLocale($locale = null)
    {
        if (Versioned::get_stage() === Versioned::LIVE) {
            return $this->isPublishedInLocale($locale);
       }
        return $this->isDraftedInLocale($locale);
    }
```

And then methods which CALL existsInLocale don't need to care about the versioning type.

Reply:
We already have `isDraftedInLocale` so I don't see the problem with existsInLocale being stage aware. :)

Reply:
I like this idea :)